aws_lambda_powertools/utilities/data_classes/common.py

```diff
@@ -1,9 +1,9 @@
 import base64
 import json
-from typing import Any, Dict, Optional
+from typing import Any, Dict, Iterator, Mapping, Optional
 
 
-class DictWrapper:
+class DictWrapper(Mapping):
```

Comment:
NOTE: I thought for a long time whether or not to explicitly add the `Mapping` protocol here or to simply implement the missing dunder methods and allow duck-typing to carry the day. 

In the end, I decided it was better to explicitly add the `Mapping` protocol because, all things being equal, I think it is bad for subclasses of this class to break that protocol. It is good for `mypy` to warn when an implementer chooses to break the protocol. It should require an explicit `type: ignore` comment when choosing to break the protocol.

There is one existing class, `StreamRecord`, that breaks the protocol. There are no subclasses of `StreamRecord` and it is used only within the context of a `DynamoDBRecord` object. Because there is only one instance of this type breakage, and it is so localized, I marked `StreamRecord.keys` as `type: ignore` and added a test to indicate that this class explicitly breaks the `Mapping` protocol.

Reply:
In older days I'd ask to remove it as we're breaking it in `StreamRecord` - I'd take this as an instance of `Practicality Beats Purity`.

It's highly unlikely a customer will use `.keys` to get the keys of a `StreamRecord` dict given the intent. However, if we do receive any customer complaint we'll consider removing the subclassing.